src/bank00.asm

```diff
@@ -998,78 +993,73 @@ drawMetaTile:
-    pop  HL                                            ;; 00:05a7 $e1
-    push BC                                            ;; 00:05a8 $c5
-    ld   A, [HL+]                                      ;; 00:05a9 $2a
-    ld   HL, wBackgroundGraphicsTileMapping            ;; 00:05aa $21 $70 $d0
-    ld   C, A                                          ;; 00:05ad $4f
-    ld   B, $00                                        ;; 00:05ae $06 $00
-    add  HL, BC                                        ;; 00:05b0 $09
-    ld   L, [HL]                                       ;; 00:05b1 $6e
-    pop  BC                                            ;; 00:05b2 $c1
-    ld   H, C                                          ;; 00:05b3 $61
-    call storeDEatBackgroundDrawPosition               ;; 00:05b4 $cd $95 $04
+    ld   A, [HL+]
+    ld   B, A
+    ld   A, [HL+]
+    ld   C, A
+    ld   A, [HL+]
+    ld   D, $00
+    ld   E, [HL]
+    push BC
+    ld   HL, wBackgroundGraphicsTileMapping
+    add  HL, DE
+    ld   C, [HL]
+    ld   HL, wBackgroundGraphicsTileMapping
+    ld   E, A
+    add  HL, DE
+    ld   B, [HL]
+    pop  DE
+    push BC
+    ld   A, D
+    ld   D, $00
+    ld   HL, wBackgroundGraphicsTileMapping
+    add  HL, DE
+    ld   C, [HL]
+    ld   HL, wBackgroundGraphicsTileMapping
+    ld   E, A
+    add  HL, DE
+    ld   B, [HL]
+    pop  DE
+    pop  HL
+    call storeBCDEinBGM
     call popBankNrAndSwitch                            ;; 00:05b7 $cd $0a $2a
     ret                                                ;; 00:05ba $c9
 
-; Exactly the same as getTileInfoPointer
-getTileInfoPointer2:
-    ld   L, A                                          ;; 00:05bb $6f
-    ld   H, $00                                        ;; 00:05bc $26 $00
-    ld   D, H                                          ;; 00:05be $54
-    ld   E, L                                          ;; 00:05bf $5d
-    add  HL, HL                                        ;; 00:05c0 $29
-    add  HL, DE                                        ;; 00:05c1 $19
-    add  HL, HL                                        ;; 00:05c2 $29
-    ld   A, [wTileDataTablePointer.High]               ;; 00:05c3 $fa $93 $d3
-    ld   D, A                                          ;; 00:05c6 $57
-    ld   A, [wTileDataTablePointer]                    ;; 00:05c7 $fa $92 $d3
-    ld   E, A                                          ;; 00:05ca $5f
-    add  HL, DE                                        ;; 00:05cb $19
-    ret                                                ;; 00:05cc $c9
+; Store BCDE at metatile with upper left corner at HL.
+; Waits until PPU mode 0 (HBlank) or 1 (VBlank).
+; Return: a = d, bc = $001f, de = de, hl = hl + 4
+storeBCDEinBGM:
+    ldh a, [rLCDC]
+    and LCDCF_ON
+    jr z, .write
+.loop:
+    ldh a, [rSTAT]
+    and STATF_BUSY
+    jr nz, .loop
+.write:
+    ld a, b
+    ld [hl+], a
+    ld [hl], c
+    ld bc, $001f
+    add hl, bc
+    ld a, d
+    ld [hl+], a
+    ld [hl], e
+    inc hl
```

Comment:
It would be one cycle faster to do:
`ld a, e`
`ld [hl+], a`

It would be two cycles faster to just delete the `inc hl`. Your choice.

Reply:
Yeah, I was trying to minimize the time with VRAM cycles, but honestly I don't think the inc HL buys us anything since it doesn't land at the start of the next metatile. I'll just remove it.